app/models/post.rb

```diff
@@ -1084,8 +1080,8 @@ def link_post_uploads(fragments: nil)
   end
 
   def update_uploads_secure_status(source:)
-    if Discourse.store.external? && SiteSetting.secure_uploads?
```

Comment:
Doesn't make a difference if `secure_uploads` is not enabled -- since if that happens a post's uploads could change from secure -> not secure